docs/integrations/airflow/extractors/default_extractors.md

```diff
@@ -0,0 +1,362 @@
+```
+
+Now, we’ll start building the facets for the `OperatorLineage` object in the `get_openlineage_facets_on_start()` method.
+
+### 1. `DataSourceDatasetFacet`
+
+The `DataSourceDatasestFacet` is a simple object, containing two fields, `name` and `uri`, which should be populated with the unique name of the data source and the URI. We’ll make two of these objects, an `input_source` to specify where the data came from, and an `output_source` to specify where the data is going.
+
+A quick note about the philosophy behind the `name` and `uri` : the `uri` is built from the `namespace` and the `name`, and each is expected to be unique with respect to its environment. This means a `namespace` should be globally unique in the OpenLineage universe, and the `name` unique within the `namespace`. The two are then concatenated to form the `uri`, so that `uri = namespace + name`. The full OpenLineage naming spec can be found [here](https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md).
+
+In our case, the input `name` will be the table we are pulling data from, `self.table`, and the `namespace` will be our `self.data_source`.
+
+```python
+input_source = DataSourceDatasetFacet(
+		name=self.table,
+		uri="://".join([self.data_source, self.table]),
+)
+```
+
+The output data source object’s `name` will always be the base path given to the operator, `self.base_path`. The `namespace` is always in GCS, so we use the OpenLineage spec’s `gs://` as the scheme and our bucket as the authority, giving us `gs://{ol_bucket}`. The `uri` is simply the concatenation of the two.
+
+```python
+if not self.bucket:
+    ol_bucket = get_env_bucket()
+else:
+    ol_bucket = self.bucket
+
+output_namespace = "gs://" + ol_bucket
+output_name = self.base_path
+output_uri = "/".join(
+    [
+        output_namespace,
+        output_name,
+    ]
+)
+
+output_source = DataSourceDatasetFacet(
+    name=output_name,
+    uri=output_uri,
+)
+```
+
+### 2. Inputs
+
+Next we’ll create the input dataset object. As we are moving data from a dataframe to GCS in this operator, we’ll make sure that we are capturing all the info in the dataframe being extracted in a `Dataset`. To create the `Dataset` object, we’ll need `namespace`, `name`, and `facets` objects. The first two are strings, and `facets` is a dictionary. Our `namespace` will come from the operator, where we use `self.data_source` again. The `name` parameter for this facet will be the table, again coming from the operator’s parameter list. The `facets` will contain two entries, our `DataSourceDatasetFacet` with key "datasource" from the previous step and a `SchemaDatasetFacet` with key "schema" that will be built with a list comprehension that creates a `SchemaField` object for each column. The `inputs` parameter to `OperatorLineage` is a list of `Dataset` objects, so we’ll end up adding a single `Dataset` object to the list later. The creation of the `Dataset` object looks as follows:
```

Comment:
Suggest adding a new paragraph before "our namespace will come from..." for readability.